resources.tf

```diff
@@ -1,8 +1,168 @@
+  user_name = "breakglass"
+}
+
+data "aws_iam_role" "ci" {
+  name = "ci"
+}
+
+data "aws_iam_role" "administrator" {
+  name = "administrator"
+}
+
+data "aws_iam_role" "aws_config" {
+  name = "aws_config"
+}
+
+data "aws_iam_role" "analytical_dataset_generator" {
+  name = "analytical_dataset_generator"
+}
+
+#Create policy document
+data "aws_iam_policy_document" "published_bucket_kms_key" {
+  
+  statement {
+    sid    = "EnableIAMPermissionsBreakglass"
+    effect = "Allow"
+
+    principals {
+      type        = "AWS"
+      identifiers = [data.aws_iam_user.breakglass.arn]
+    }
+
+    actions   = ["kms:*"]
+    resources = ["*"]
+  }
+
+  statement {
+    sid       = "EnableIAMPermissionsCI"
+    effect    = "Allow"
+    actions   = ["kms:*"]
+    resources = ["*"]
+
+    principals {
+      identifiers = [data.aws_iam_role.ci.arn]
+      type        = "AWS"
+    }
+  }
+
+  statement {
+    sid    = "DenyCIEncryptDecrypt"
+    effect = "Deny"
+
+    principals {
+      type        = "AWS"
+      identifiers = [data.aws_iam_role.ci.arn]
+    }
+
+    actions = [
+      "kms:Encrypt",
+      "kms:Decrypt",
+      "kms:ImportKeyMaterial",
+      "kms:ReEncryptFrom",
+    ]
+
+    resources = ["*"]
+  }
+
+  statement {
+    sid    = "EnableIAMPermissionsAdministrator"
+    effect = "Allow"
+
+    principals {
+      type        = "AWS"
+      identifiers = [data.aws_iam_role.administrator.arn]
+    }
+
+    actions = [
+      "kms:Describe*",
+      "kms:List*",
+      "kms:Get*",
+      "kms:Put*",
+      "kms:Create*"
+    ]
+
+    resources = ["*"]
+  }
+
+  statement {
+    sid    = "EnableAWSConfigManagerScanForSecurityHub"
+    effect = "Allow"
+
+    principals {
+      type        = "AWS"
+      identifiers = [data.aws_iam_role.aws_config.arn]
+    }
+
+    actions = [
+      "kms:Describe*",
+      "kms:Get*",
+      "kms:List*"
+    ]
+
+    resources = ["*"]      
+  }
+
+  statement {
+    sid    = "EnableIAMPermissionsAnalyticDatasetGen"
+    effect = "Allow"
+    
+    principals {
+      type        = "AWS"
+      identifiers = [aws_iam_role.analytical_dataset_generator.arn]
+    }
+
+    actions = [
+     "kms:Encrypt",
+     "kms:Decrypt",
+     "kms:ReEncrypt*",
+     "kms:GenerateDataKey*",
+     "kms:DescribeKey"
+    ]
+
+    resources = ["*"]
+    
+  }
+
+  statement {
+    sid    = "EnableIAMPermissionsPdm"
+    effect = "Allow"
+
+    principals {
+      type        = "AWS"
+      identifiers = [aws_iam_role.pdm.arn]
+    }
+
+    actions = [
+     "kms:Encrypt",
+     "kms:Decrypt",
+     "kms:ReEncrypt*",
+     "kms:GenerateDataKey*",
+     "kms:DescribeKey"
+    ]
+
+    resources = ["*"]
+  }
+}
+
+# Add policy to UCFS bucket
 resource "aws_kms_key" "published_bucket_cmk" {
   description             = "UCFS published Bucket Master Key"
   deletion_window_in_days = 7
   is_enabled              = true
   enable_key_rotation     = true
+  depends_on              = [aws_iam_role.pdm]
```

Comment:
This `depends_on` can be removed; a key policy is attached to a key, not to an IAM role.